src/compile.rs

```diff
@@ -101,11 +105,79 @@ fn compile_universal2(
     Ok(result)
 }
 
+/// We want to use `zig cc` as linker and c compiler. We want to call `python -m ziglang cc`, but
+/// cargo only accepts a path to an executable as linker, so we add a wrapper script. We then also
+/// use the wrapper script to pass arguments and substitute an unsupported argument.
+///
+/// We create different files for different args because otherwise cargo might skip recompiling even
+/// if the linker target changed
+fn prepare_zig_linker(context: &BuildContext) -> Result<PathBuf> {
+    let target = &context.target;
+    let arch = if target.cross_compiling() {
+        if matches!(target.target_arch(), Arch::Armv7L) {
+            "armv7".to_string()
+        } else {
+            target.target_arch().to_string()
+        }
+    } else {
+        "native".to_string()
+    };
+    let (zig_linker, cc_args) = match context.platform_tag {
+        // Not sure branch even has any use case, but it doesn't hurt to support it
+        None | Some(PlatformTag::Linux) => {
+            ("./zigcc-gnu.sh".to_string(), format!("{}-linux-gnu", arch))
+        }
+        Some(PlatformTag::Musllinux { x, y }) => {
+            println!("⚠️  Warning: zig with musl is unstable");
+            (
+                format!("./zigcc-musl-{}-{}.sh", x, y),
+                format!("{}-linux-musl", arch),
+            )
+        }
+        Some(PlatformTag::Manylinux { x, y }) => (
+            format!("./zigcc-gnu-{}-{}.sh", x, y),
+            // https://github.com/ziglang/zig/issues/10050#issuecomment-956204098
+            format!(
+                "${{@/-lgcc_s/-lunwind}} -target {}-linux-gnu.{}.{}",
+                arch, x, y
+            ),
+        ),
+    };
+
+    let zig_linker_dir = dirs::cache_dir()
+        // If the really is no cache dir, cwd will also do
+        .unwrap_or_else(|| PathBuf::from("."))
+        .join(env!("CARGO_PKG_NAME"))
+        .join(env!("CARGO_PKG_VERSION"));
+    fs::create_dir_all(&zig_linker_dir)?;
+    let zig_linker = zig_linker_dir.join(zig_linker);
+
+    #[cfg(target_family = "unix")]
+    let mut custom_linker_file = OpenOptions::new()
+        .create(true)
+        .write(true)
+        .truncate(true)
+        .mode(0o700)
+        .open(&zig_linker)?;
+    #[cfg(not(target_family = "unix"))]
+    let mut custom_linker_file = File::create(&zig_linker)?;
+    writeln!(&mut custom_linker_file, "#!/bin/bash")?;
```

Comment:
Hmm, this doesn't work on Windows apparently.

Reply:
Here is an idea, can we provide a `zig` subcommand in maturin instead so it's easier to support non-unix platforms? 

* maturin zig cc
* maturin zig c++

And it should be made compatible with both plain zig and the python ziglang package.

Reply:
I haven't check if this also applies to windows, but on linux the linker needs to be the name or path to a binary, i.e. no cli arguments, so a subcommand wouldn't help. If there's the same rule on windows we could write a batch file or something, otherwise do `CC="python -m ziglang cc"` directly. An alternative could be writing python scripts, but again I don't know how well windows like python scripts as linkers.

Reply:
Windows support would be nice, but isn't a blocker for merging imho

Reply:
Agreed windows support is not a blocker. 

I proposed maturin subcommand to simplify replacing `libgcc_s` with `libunwind` because I don't know how to do that in windows batch, a shell/batch script is still needed but it just forwards arguments to a maturin subcommand. 🤪

Reply:
Oh for that it makes total sense!